src/Schema/TableDiff.php

```diff
@@ -17,42 +16,31 @@ class TableDiff
      *
      * @internal The diff can be only instantiated by a {@see Comparator}.
      *
-     * @param array<Column>              $addedColumns
-     * @param array<ColumnDiff>          $changedColumns
-     * @param array<Column>              $removedColumns
-     * @param array<Index>               $addedIndexes
-     * @param array<Index>               $changedIndexes
-     * @param array<Index>               $removedIndexes
-     * @param list<ForeignKeyConstraint> $addedForeignKeys
-     * @param list<ForeignKeyConstraint> $changedForeignKeys
-     * @param list<ForeignKeyConstraint> $removedForeignKeys
-     * @param array<string, Column>      $renamedColumns
-     * @param array<string, Index>       $renamedIndexes
+     * @param array<ForeignKeyConstraint> $droppedForeignKeys
+     * @param array<Column>               $addedColumns
+     * @param array<ColumnDiff>           $modifiedColumns
+     * @param array<Column>               $droppedColumns
+     * @param array<string, Column>       $renamedColumns
+     * @param array<Index>                $addedIndexes
+     * @param array<Index>                $modifiedIndexes
+     * @param array<Index>                $droppedIndexes
+     * @param array<string, Index>        $renamedIndexes
+     * @param array<ForeignKeyConstraint> $addedForeignKeys
```

Comment:
I wish there was a way to express that this is a set explicitely (`list` is great at that but does have an extra constraint we don't need). Now it looks like we forgot to document the keys.

Reply:
We will likely introduce a dedicated data structure for these collections to address https://github.com/doctrine/dbal/issues/4772. Currently, table constraints and indexes are indexed by the lower-case name but this doesn't work for case-sensitive platforms.